recipes/filter-api-validator.rb

```diff
@@ -0,0 +1,35 @@
+#
+# Cookbook Name:: repose
+# Recipe:: filter-header-normalization
+#
+# Copyright (C) 2013 Rackspace Hosting
+#
+
+include_recipe 'repose::install'
+
+unless node['repose']['filters'].include? 'api-validator'
+  filters = node['repose']['filters'] + ['api-validator']
+  node.normal['repose']['filters'] = filters
+end
+
+template "#{node['repose']['config_directory']}/validator.cfg.xml" do
+  owner node['repose']['owner']
+  group node['repose']['group']
+  mode '0644'
+  variables(
+    :multi_role_match => node['repose']['api_validator']['multi_role_match'],
+    :delegating_quality => node['repose']['api_validator']['delegating_quality'],
+    :validators => node['repose']['api_validator']['validators']
+  )
+  notifies :restart, 'service[repose-valve]'
+end
+
+node['repose']['api_validator']['validators'].each do |f|
+  if f.has_key?("wadl") 
+    wfile = f['wadl']
+    cookbook_file wfile do
```

Comment:
There is a cookbook attribute on the cookbook_file resource that allows you to use a file in another cookbook.

http://docs.chef.io/resource_cookbook_file.html#attributes

If we set it to `node['repose']['api_validator']['wadl_cookbook']` and make the default `repose` it will allow others to bring there own wadl files in another cookbook without having to modify this one.